components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -884,6 +884,24 @@ private void callDefaultProvisioningHandler(String username, AuthenticationConte
         // Remove role claim from local claims as roles are specifically handled.
         localClaimValues.remove(FrameworkUtils.getLocalClaimUriMappedForIdPRoleClaim(externalIdPConfig));
         localClaimValues.remove(UserCoreConstants.USER_STORE_GROUPS_CLAIM);
+
+        Map<String, String> runtimeClaims = context.getRuntimeClaims();
+        Map<String, String> remoteClaims =
+                (Map<String, String>) context.getProperty(FrameworkConstants.UNFILTERED_IDP_CLAIM_VALUES);
+
+        // Remove or revert runtime claims from local claims before calling JIT provisioning.
+        for (Map.Entry<String, String> entry : runtimeClaims.entrySet()) {
+            String localClaimURI = entry.getKey();
+            if (claimMapping != null && remoteClaims.containsKey(claimMapping.get(localClaimURI))) {
+                // If remote claim value was overridden by temp claim, revert it
+                String claimValue = remoteClaims.get(claimMapping.get(localClaimURI));
+                localClaimValues.put(localClaimURI, claimValue);
+            } else {
+                // If no remote claim value was overridden by temp claim, remove it
+                localClaimValues.remove(localClaimURI);
```

Comment:
As per my understanding, we are removing the temp claims from the `localCliamValues` if it is not there in the `remoteClaims` set. Hence the removal part of temp claims from the `localCliamValues` will affect the following behaviour:
-  keep a temporary claim value for the session and pass to the SP?

Reply:
> Tested this scenario in the following setups.

> Here, `pincode` is a newly created attribute.

### Asgardeo Dev Console
- IDP: `Google`
- IDP JIT Config:
```
"jit": {
    "isEnabled": true,
    "scheme": "PROVISION_SILENTLY",
    "userstore": "DEFAULT",
    "associateLocalUser": true,
    "attributeSyncMethod": "PRESERVE_LOCAL”
},
```
- Adaptive Script:
```
subject = context.currentKnownSubject;
subject.claims["http://wso2.org/claims/pincode"] = "5789"
```
- Results: `Both session pincode and the pincode in the user profile got changed` (Not the expected the behavior)

### Asgardeo Local Setup (with the fix)
- IDP: `Google`
- IDP JIT Config:
```
"jit": {
    "isEnabled": true,
    "scheme": "PROVISION_SILENTLY",
    "userstore": "DEFAULT",
    "associateLocalUser": true,
    "attributeSyncMethod": "PRESERVE_LOCAL”
},
```
- Adaptive Script:
```
subject = context.currentKnownSubject;
subject.claims["http://wso2.org/claims/pincode"] = "5789"
```
- Results: `Session picode changed to the new value defined by the temp claim. User profile picode remained the same.` (expected behavior with the fix)